packages/kbn-test/src/kbn_client/kbn_client_requester_error.ts

```diff
@@ -7,15 +7,21 @@
  */
 
 import { AxiosError } from 'axios';
-
 export class KbnClientRequesterError extends Error {
   axiosError?: AxiosError;
   constructor(message: string, error: unknown) {
     super(message);
     this.name = 'KbnClientRequesterError';
 
     if (error instanceof AxiosError) {
-      this.axiosError = error;
+      this.axiosError = clean(error);
     }
   }
 }
+function clean(error: Error): AxiosError {
+  const _ = AxiosError.from(error);
```

Comment:
Can we move this into positive form to make sure nothing slips through?  Maybe return only the message field.

Reply:
I can't think of why not off hand.  Let me run it locally and see.

Reply:
<img width="1728" alt="Screenshot 2024-07-15 at 17 02 01" src="https://github.com/user-attachments/assets/324f2811-76a3-4857-b2d8-0be13661db30">

@jbudz I reckon we're striking out on message.  Does any other field look good?

Reply:
fyi @jbudz  https://www.npmjs.com/package/axios-error-redact

Reply:
response.data may be what we're looking for.

Reply:
Lemme check it.

Reply:
Yeah it's not in scope.  That's what we get for using try / catch for control flow :(

Reply:
> We don't know if Axios error will have updates in the future, I agree that it would be easier if we explicitly pick only the props that we need.

@dmlemeshko I get the feeling you regarded the link I posted as a suggestion we should use.  That's my fault for being unclear.  I was only showing that we are not the only folks with this pain :)

Reply:
@jbudz what do you think now?

Reply:
What's being returned with all the fields deleted?  Can we just return that?

Reply:
https://github.com/elastic/kibana/pull/171803/files#diff-e6748a42825d4ec036e06ea06dedb1506b13e16eccd36d8c63e5fceb3d1ba642R204 should be fine.

Reply:
https://github.com/elastic/kibana/pull/188292/commits/69de38d6425830f29abd8d05850fe881a7a5eb3c

Reply:
69d38d was reverted.  Context?

Reply:
Yeah, there were several dependencies upon the computation of the error message throwing an error.
In other words, there are several call sites that depend on that error being thrown, which has nothing to do with redaction of auth info.

Reply:
But, I can put the error object back to how it was to show how that caused errors.

Reply:
That's okay.  Lets get this merged, the redaction as is is still important and the error can be addressed after.

Reply:
Ok, I'll revert and merge once it's passed